docs/compute-resources.md

```diff
@@ -52,6 +52,251 @@ requirements for `Step`containers, they must be treated as if they are running i
 Tekton adjusts `Step` resource requirements to comply with [LimitRanges](#limitrange-support).
 [ResourceQuotas](#resourcequota-support) are not currently supported.
 
+Kubernetes interprets `Pod` resources requirements by taking the value summed up by each container's resources requirements. It leads Tekton to hoard resources computed by all `Steps`(containers), which can only be used on each sequentially executed `Step`. This observation forms the interest to support a `Task`/`Pod` level resources requirements, which naturally matches Kubernetes computing logic and can help users save significantly on requested resources.
+
+`Task`-level resources requirements can be configured in `Task.spec.resources` or `PipelineRun.spec.taskRunSpecs.resources`.
+
+e.g.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: foo 
+spec:
+  steps:
+    - name: foo1
+    - name: foo2
+  resources:
+    requests:
+      cpu: 1 
+    limits:
+      cpu: 2
+```
+
+### Applying Resources Requirements
```

Comment:
Some of this commentary could be reworded to focus less on the implementation and more on the user experience.
For example, where you talk about limits being enforced by the container runtime, or validation being performed by the webhook.

Reply:
closed - cleaned